pkg/cableengine/cableengine.go

```diff
@@ -63,16 +63,16 @@ type Engine interface {
 type engine struct {
 	sync.Mutex
 	driver              cable.Driver
-	localCluster        types.SubmarinerCluster
-	localEndpoint       types.SubmarinerEndpoint
+	localCluster        *types.SubmarinerCluster
+	localEndpoint       *types.SubmarinerEndpoint
```

Comment:
I think it's better to store by value and make a copy of the structs in `NewEngine` so any changes made by the caller afterwards aren't observed here.